client.ts

```diff
@@ -0,0 +1,463 @@
+        .then(({ allowed, $response: response }) => {
+          const result = {
+            allowed: allowed || false,
+            _request: tuple,
+          };
+          setNotEnumerableProperty(result, "$response", response);
+          return result;
+        })
+        .catch(err => ({
+          error: err,
+          _request: tuple,
+        }))
+      )), batchCheckRequest, DEFAULT_MAX_METHOD_PARALLEL_REQS).then(results => results.flat());
+  }
+
+  /**
+   * Expand - Expands the leaves that have a particular relation to an object (evaluates)
+   * @param expandRequest
+   * @param {string} expandRequest.relation The relation
+   * @param {string} expandRequest.object The object, must be of the form: `<type>:<id>`
+   * @param options
+   */
+  async expand(expandRequest: ClientExpandRequest, options: ClientRequestOptsWithAuthZModelId = {}): PromiseResult<ExpandResponse> {
+    return this.openFgaApi.expand({
+      authorization_model_id: this.getAuthorizationModelId(options),
+      tuple_key: expandRequest,
+    });
+  }
+
+  /**
+   * ListObjects - List the objects of a particular type that the user has a certain relation to (evaluates)
+   * @param listObjectsRequest
+   * @param options
+   */
+  async listObjects(listObjectsRequest: ClientListObjectsRequest, options: ClientRequestOptsWithAuthZModelId = {}): PromiseResult<ListObjectsResponse> {
+    const authorizationModelId = this.getAuthorizationModelId(options);
+    return this.openFgaApi.listObjects({
+      authorization_model_id: authorizationModelId,
+      user: listObjectsRequest.user,
+      relation: listObjectsRequest.relation,
+      type: listObjectsRequest.type,
+      contextual_tuples: { tuple_keys: listObjectsRequest.contextual_tuples || [] },
+    }, options);
+  }
+
+  /**
+   * ListRelations - Lists all the relations a user has with an object (evaluates)
+   * @param {object} listRelationsRequest
+   * @param {string} listRelationsRequest.user The user object, must be of the form: `<type>:<id>`
+   * @param {string} listRelationsRequest.object The object, must be of the form: `<type>:<id>`
+   * @param {string[]} [listRelationsRequest.relations] The list of relations to check, if not sent default to the all those for that type int the model
+   * @param options
+   */
+  async listRelations(listRelationsRequest: ClientListRelationsRequest, options: ClientRequestOptsWithAuthZModelId = {}): Promise<ClientListRelationsResponse> {
+    const { user, object } = listRelationsRequest;
+    let { relations } = listRelationsRequest;
+    
+    // 1- Get the list of relations on that object type
+    // 2- Filter out/error out on invalid relations
+    // 2- Call check in batch
+    const authorizationModel: AuthorizationModel | undefined = this.getAuthorizationModelId(options) ?
+      (await this.readAuthorizationModel()).authorization_model : (await this.readLatestAuthorizationModel(options)).authorization_model;
```

Comment:
Can you verify whether we need to pass in the authorization model id/options to this.readAuthorizationModel?